src/wasm_module.rs

```diff
@@ -0,0 +1,35 @@
+use std::sync::{Arc, RwLock};
+
+use wasi_common::pipe::{ReadPipe, WritePipe};
+use wasmtime::*;
+use wasmtime_wasi::*;
+
+// In future this might be pre-instantiated or something like that, so we will
+// just abstract it to be safe.
+#[derive(Clone, Debug)]
+pub enum WasmModuleSource {
+    Blob(Arc<Vec<u8>>),
+}
```

Comment:
@radu-matei Does it kinda make sense how you could add something here to add support for pre-inited modules?